File: play-2.4.x/K-Modules/kmodules/K-Base/app/K/Aop/annotations/CheckCaptcha.java
package K.Aop.annotations;

import play.mvc.With;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Created by kk on 15/1/4.
 */
@With(K.Aop.actions.CheckCaptchaAction.class)
@Target({ElementType.TYPE, ElementType.METHOD})
@Retention(RetentionPolicy.RUNTIME)
public @interface CheckCaptcha {
    String captcha_key_arg() default "captcha_key";

    String captcha_code_arg() default "captcha_code";
}

File: play-kotlin-2.5.x/PlayKotlin/kmodules/K-Base/app/K/Aop/annotations/CheckCaptcha.kt
package K.Aop.annotations

import play.mvc.With
import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

/**
 * Created by kk on 15/1/4.
 */
@With(K.Aop.actions.CheckCaptchaAction::class)
@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE, AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@Retention(RetentionPolicy.RUNTIME)
annotation class CheckCaptcha(val captcha_key_arg: String = "captcha_key", val captcha_code_arg:

String = "captcha_code")
